Source/Python/WarpX_py.H

```diff
@@ -13,19 +13,7 @@
 
 extern "C" {
 
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_afterinit;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_beforeEsolve;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_poissonsolver;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_afterEsolve;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_beforedeposition;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_afterdeposition;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_particlescraper;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_particleloader;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_beforestep;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_afterstep;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_afterrestart;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_particleinjection;
-    extern WARPX_CALLBACK_PY_FUNC_0 warpx_py_appliedfields;
+    extern std::map< std::string, WARPX_CALLBACK_PY_FUNC_0 > warpx_callback_py_map;
```

Comment:
Oh, actually... is that possible? An `extern` with stable ABI for a C++ type? :-o

Reply:
I think I misunderstood what you planned on Slack, I am not sure how well supported/stable this is

Reply:
The map doesn't need C linkage so I removed that. Are there other concerns with using `extern` here?

Reply:
Got it, yes linkage looks clean now :)